lib/node_modules/@stdlib/math/base/special/gcd/src/main.c

```diff
@@ -0,0 +1,181 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2024 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "stdlib/math/base/special/gcd.h"
+#include "stdlib/math/base/assert/is_nan.h"
+#include "stdlib/math/base/assert/is_integer.h"
+#include "stdlib/constants/float64/pinf.h"
+#include "stdlib//constants/float64/ninf.h"
+#include "stdlib/constants/int32/max.h"
+#include <stdint.h>
+
+/**
+* Computes the greatest common divisor (gcd) using the binary GCD algorithm.
+*
+* @param a    input value
+* @param b    input value
+* @return	  output value
+*
+* @example
+* double out = largeIntegers( 1.2676506002282294e+30, 9007199254740992 );
+* // returns 9007199254740992.0
+*/
+
+double largeIntegers( const double a, const double b ) {
+	int64_t ac;
+	int64_t bc;
+	int64_t k;
+	int64_t t;
+
+	ac = (int64_t)a;
+	bc = (int64_t)b;
```

Comment:
For large integers, you don't need to cast to `int64_t`. All the operations below (`%`, `/`) are defined for double-precision floating-point numbers. This should be refactored accordingly. `k` and `t` can be doubles.

Reply:
Correction: you can use `fmod` for computing the remainder for double-precision numbers.